### PR TITLE
one-level view frustum culling

### PR DESCRIPTION
**Pull request by barlebino**

### CODE REVIEW

File: src/Engine/Component/CameraComponents/CameraController.cpp [hunk 1]
```diff
@@ -4,33 +4,52 @@
 #include "IO/Keyboard.hpp"
 #include "Component/SpatialComponents/SpatialComponent.hpp"
 
+// Debug
+#include <stdio.h>
```
Comment (jaafersheriff) [inline src/Engine/Component/CameraComponents/CameraController.cpp]: Can this be removed?

File: src/Engine/Component/CameraComponents/CameraController.cpp [hunk 2]
```diff
@@ -4,33 +4,52 @@
...
+
 void CameraController::init() {
 
 }
 
 void CameraController::update(float dt) {
+    /* Checks if the camera direction or location has been modified */
+    bool modified;
```
Comment (jaafersheriff) [inline src/Engine/Component/CameraComponents/CameraController.cpp]: This boolean doesn't seem to do anything. Also, this logic should go in the Camera, not the CameraController. Also, I already added this logic to Camera.

File: src/Engine/Component/CameraComponents/CameraController.cpp [hunk 3]
```diff
@@ -4,33 +4,52 @@
...
     }
     if (Keyboard::isKeyPressed(down)) {
         moveDown(dt);
+        modified = true;
     }
+
+    // TODO: Update frustum data only when you move the camera
+    /* If the camera has been modified, then update the view frustum */ 
+    /* if (modified) {
+        printf("Camera has been modified\n");
```
Comment (jaafersheriff) [inline src/Engine/Component/CameraComponents/CameraController.cpp]: Can this be removed?

File: src/Engine/Component/Component.hpp [hunk 4]
```diff
@@ -13,6 +13,7 @@ class Component {
         virtual void update(float);
 
         GameObject* getGameObject() { return gameObject; }
+        const GameObject* getGameObject() const { return gameObject; }
```
Comment (jaafersheriff) [inline src/Engine/Component/Component.hpp]: Duplicated function

File: src/Engine/Loader/Loader.hpp [hunk 5]
```diff
@@ -8,6 +8,7 @@
 #include "Library.hpp"
 
 #include <cstdint>
+#include <iostream>
```
Comment (jaafersheriff) [inline src/Engine/Loader/Loader.hpp]: iostream isn't used in Loader.hpp, and it's already included in the .cpp

File: src/Engine/Shaders/DiffuseShader.cpp [hunk 6]
```diff
@@ -112,4 +136,6 @@ void DiffuseShader::render(const std::string & name, const std::vector<Component
             glBindTexture(GL_TEXTURE_2D, 0);
         }
     }
+
+    printf("Rendered %d DiffuseRenderComponents\n", renderCount);
```
Comment (jaafersheriff) [inline src/Engine/Shaders/DiffuseShader.cpp]: Can we remove all the debug stuff?

File: src/Engine/System/RenderSystem.hpp [hunk 7]
```diff
@@ -9,6 +9,7 @@
 
 #include <map>
 #include <string>
+#include <iostream>
```
Comment (jaafersheriff) [inline src/Engine/System/RenderSystem.hpp]: iostream shouldn't be included in headers

File: src/Engine/Component/CameraComponents/CameraController.cpp (hunk 1 above)
Comment (barlebino) [reply src/Engine/Component/CameraComponents/CameraController.cpp]: Yes

File: src/Engine/Component/CameraComponents/CameraController.cpp (hunk 3 above)
Comment (barlebino) [reply src/Engine/Component/CameraComponents/CameraController.cpp]: Yes

File: src/Engine/Component/CameraComponents/CameraController.cpp (hunk 2 above)
Comment (barlebino) [reply src/Engine/Component/CameraComponents/CameraController.cpp]: Ok. This can be removed

File: src/Engine/Component/Component.hpp (hunk 4 above)
Comment (barlebino) [reply src/Engine/Component/Component.hpp]: This can be removed

File: src/Engine/Loader/Loader.hpp (hunk 5 above)
Comment (barlebino) [reply src/Engine/Loader/Loader.hpp]: This can be removed

File: src/Engine/Shaders/DiffuseShader.cpp (hunk 6 above)
Comment (barlebino) [reply src/Engine/Shaders/DiffuseShader.cpp]: Yes

File: src/Engine/System/RenderSystem.hpp (hunk 7 above)
Comment (barlebino) [reply src/Engine/System/RenderSystem.hpp]: Removing this line gives "'cerr' is not a member of 'std". Where should we place this line?

**Pull request closed without merging**